tests/core/middleware/test_filter_middleware.py

```diff
@@ -0,0 +1,136 @@
+import pytest
+
+from web3 import Web3
+from web3.middleware import (
+    construct_result_generator_middleware,
+    local_filter_middleware,
+)
+from web3.middleware.filter import (
+    iter_latest_block_ranges,
+    block_range_chunker,
+)
+from web3.providers.base import (
+    BaseProvider,
+)
+
+
+class DummyProvider(BaseProvider):
+    def make_request(self, method, params):
+        raise NotImplementedError("Cannot make request for {0}:{1}".format(
+            method,
+            params,
+        ))
+
+
+@pytest.fixture(scope='function')
```

Comment:
Not highly opposed to this but `"function"` is the default `scope` for `pytest.fixture`